### PR TITLE
Bugfix/azurerm downgrade

### PR DESCRIPTION
**Pull request by JohnNKing**
# azurerm Downgrade and Terraform Deployment Testing

Temporarily downgrading the the azurerm Terraform provider to fix a bug preventing deployments.

Also adding a Terraform test deployment workflow to prevent future PRs from breaking deployments.  This involves adding a new reusable GitHub Action workflow that deploys the Terraform.  We use it for the this PR flow and for our original staging deployment.  This test deployment only happens when we edit anything in the `/operations/` folder.

## Issue

#79.

### CODE REVIEW

**Review by JohnNKing — COMMENTED** [review]
One other thought: Should terraform-ci.yml go away? If terraform-ci-deploy.yml also were to do a TF format check, then there's really no need for it.

File: .github/workflows/terraform-ci-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            operations:
+              - 'operations/**'
```
[the remark this answers is not in the document]
Comment (JohnNKing) [reply .github/workflows/terraform-ci-deploy.yml:25]: Ah -- that's right. I forgot where the TF provider version is specified. 🤦‍♂️

**Review by JohnNKing — COMMENTED** [review]
Looks great! I'd approve it, but I guess I can't as I originally created the PR. 😆

**Comment by halprin** [comment]
Ah, yeah.  Hah!  I'll approve, and you can merge.

**Review by halprin — APPROVED** [review]

**Pull request merged**